specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2021-03-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,1956 @@
+      "required": [
+        "location",
+        "sku"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "identity": {
+          "$ref": "#/definitions/ResourceIdentity",
+          "description": "The managed identity information, if configured."
+        },
+        "properties": {
+          "$ref": "#/definitions/ConfigurationStoreProperties",
+          "description": "The properties of a configuration store.",
+          "x-ms-client-flatten": true
+        },
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The sku of the configuration store."
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "Resource system metadata."
+        }
+      }
+    },
+    "ConfigurationStoreProperties": {
+      "description": "The properties of a configuration store.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "description": "The provisioning state of the configuration store.",
+          "enum": [
+            "Creating",
+            "Updating",
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Canceled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "creationDate": {
+          "format": "date-time",
+          "description": "The creation date of configuration store.",
+          "type": "string",
+          "readOnly": true
+        },
+        "endpoint": {
+          "description": "The DNS endpoint where the configuration store API will be available.",
+          "type": "string",
+          "readOnly": true
+        },
+        "encryption": {
+          "$ref": "#/definitions/EncryptionProperties",
+          "description": "The encryption settings of the configuration store."
+        },
+        "privateEndpointConnections": {
+          "description": "The list of private endpoint connections that are set up for this resource.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnectionReference"
+          }
+        },
+        "publicNetworkAccess": {
+          "description": "Control permission for data plane traffic coming from public networks while private endpoint is enabled.",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PublicNetworkAccess",
+            "modelAsString": true
+          }
+        },
+        "disableLocalAuth": {
```

Comment:
>disableLocalAuth [](start = 9, length = 16)

rather than having boolean properties - the recommendation is to use enum properties instead. That allows you to have more flexibiliy in the future when you want to disable multiple auth types eg - local and dsts - you don't need to add a bool every time. So instead of having a 'disableLocalAuth' property you would have a 'supportedAuthTypes' enum property for example.

Reply:
https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean - search for boolean here

Reply:
@varunkumta

This API is not designed by us, but is instead led by JEDI. All JEDI services are required to add a bool property named `disableLocalAuth`.